core/iwasm/fast-jit/fe/jit_emit_numberic.c

```diff
@@ -945,9 +957,15 @@ compile_int_shru(JitCompContext *cc, JitReg left, JitReg right, bool is_i32)
 
     res = is_i32 ? jit_cc_new_reg_I32(cc) : jit_cc_new_reg_I64(cc);
 #if defined(BUILD_TARGET_X86_64) || defined(BUILD_TARGET_AMD_64)
-    GEN_INSN(MOV, is_i32 ? ecx_hreg : rcx_hreg, right);
-    GEN_INSN(SHRU, res, left, is_i32 ? ecx_hreg : rcx_hreg);
-    GEN_INSN(MOV, ecx_hreg, ecx_hreg);
+    if (jit_reg_is_const(right)) {
```

Comment:
should move `if` branch to `DEF_UNI_INT_CONST_OPS(shru)`

Reply:
Done